ast/compile_test.go

```diff
@@ -1594,48 +1587,85 @@ func TestCompilerCheckUnusedImports(t *testing.T) {
 			`,
 		},
 		{
-			note: "pass_case_5",
+			note: "import used as function operand",
 			module: `package p
 			import data.foo
 			r = count(foo) > 1 # only one operand
 			`,
 		},
 		{
-			note: "pass_case_6",
+			note: "import used as function operand, compount term",
 			module: `package p
 			import data.foo
-			r = sprintf("%v %d", [foo, 0]) # more complicated operands
+			r = sprintf("%v %d", [foo, 0])
 			`,
 		},
 		{
-			note: "pass_case_7",
+			note: "import used as plain term",
 			module: `package p
 			import data.foo
 			r {
-				foo # not a call, plain term
+				foo
 			}
 			`,
 		},
 		{
-			note: "pass_case_8",
+			note: "import used in 'every' domain",
 			module: `package p
 			import future.keywords.every
 			import data.foo
 			r {
-				every x in foo { x > 1 } # import in 'every' domain
+				every x in foo { x > 1 }
 			}
 			`,
 		},
 		{
-			note: "pass_case_9",
+			note: "import used in 'every' body",
 			module: `package p
 			import future.keywords.every
 			import data.foo
 			r {
-				every x in [1,2,3] { x > foo } # import in 'every' body
+				every x in [1,2,3] { x > foo }
 			}
 			`,
 		},
+		{
+			note: "future import kept even if unused",
+			module: `package p
+			import future.keywords
+
+			r { true }
+			`,
+		},
+		{
+			note: "shadowed var name in function arg",
+			module: `package p
+			import data.foo # unused
+
+			r { f(1) }
+			f(foo) = foo == 1
+			`,
+			expectedErrors: Errors{
+				&Error{
+					Location: NewLocation([]byte("import"), "", 2, 4),
+					Message:  "import data.foo unused",
+				},
+			},
+		},
+		{
+			note: "shadowed assigned var name",
+			module: `package p
+			import data.foo # unused
+
+			r { foo := true; foo }
+			`,
+			expectedErrors: Errors{
+				&Error{
+					Location: NewLocation([]byte("import"), "", 2, 4),
+					Message:  "import data.foo unused",
+				},
+			},
+		},
```

Comment:
Should we also add test cases for when an import is used/not used as a rule value? E.g.
```rego
package p
import data.foo # unused
import data.bar

r = bar { true }
```
and
```rego
package p
import data.foo # unused
import data.bar #unused

r = data.bar { true }
```